Source/Classes/AnimatedImages/PINCachedAnimatedImage.m

```diff
@@ -395,13 +425,16 @@ - (void)setCoverImageReadyCallback:(PINAnimatedImageInfoReady)coverImageReadyCal
  */
 - (void)clearAnimatedImageCache
 {
-    [_lock lockWithBlock:^{
-        _coverImage = nil;
-        [_cachedOrCachingFrames enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL * _Nonnull stop) {
-            [_frameCache removeObjectForKey:@(idx)];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
It would be ideal if these dispatches first checked if on the main queue, and if not, ran synchronously - in general that would provide a bit more control & less overhead for apps that want to drive these calls.